scripts/release.sh

```diff
@@ -0,0 +1,281 @@
+    echo
+    echo "Current version of $submodule_name: "$submodule_current_version
+    echo "Latest version of $submodule_name: "$submodule_latest_version
+    echo "Please update the submodule $submodule_name before continuing with the release."
+    echo
+    echo "You must also edit the dependancy information in SmartDeviceLink-iOS.podspec"
+    echo
+fi
+
+
+# 5 update changelog
+# IDEA - I should record the timestamp of the changelog, and then check after the user returns to see that they did touch the file.
+# IDEA - we could also insert a template into the changelog that includes the version the users have selected above.
+echo 
+echo "Please go update Changelog.md, then return here and press enter."
+read user_input
+# Changelog.md
+
+# 6 generate documentation
+echo 
+echo "6 generate documentation"
+echo "Would you like to automatically generate documentation with Jazzy (Y/n)?"
+read user_input
+if [[ ! $user_input == [Nn] ]]; then    
+    # 6 Install Jazzy
+    # Check if Jazzy is already installed, and if not then install jazzy
+    echo "Check if Jazzy is already installed, and install if necessary (may require password)"
+    if [ -z "$(mdfind -name 'Jazzy')" ]; then
+        echo "6 Install Jazzy"
+        sudo gem install jazzy
+    fi
+
+    # this runs Jazzy to generate the documentation
+    echo "Run Jazzy to generate documentation"
+    jazzy --clean --objc --framework-root SmartDeviceLink --sdk iphonesimulator --umbrella-header SmartDeviceLink/public/SmartDeviceLink.h --theme theme --output docs
+#else
+#    echo "not Jazzy..."
+fi
+
+# 7 ensure RPC_SPEC has released to master
+echo
+echo "Step 7: ensure RPC_SPEC has released to master"
+# TODO - cleanup wording
+# IDEA - maybe provide links to assist?
+echo "Please ensure RPC_SPEC has released to master and return here"
+read user_input
+
+# 7 update the submodule to point to new release
+#echo "update the submodule to point to new release"
+# TODO - cleanup wording
+# IDEA - maybe provide links to assist?
+echo
+echo "Please update the submodule to point to new release and return here"
+read user_input
+
+
+echo
+echo "Please perform the following steps"
+# and every step of this process, the user needs complete control.
+# that means we tell them what we want to do, ask them if they want us to do it, report the results, and gracefully handle exits.
+
+# git commands
+#todo - I don't like any of how this section is handled.
+# 8 commit release to develop
+echo "step 8: commit release to develop"
+# git checkout <branch>
+# git add -A
+# git commit -m <message here>
+# git push --set-upstream origin <branchname>
+
+# 8 merge release to master
+echo "8.1 merge release to master"
+# git merge <branch> <master>
+
+# 8 tag it
+echo "8.2 tag the release"
+# git tag <something>
+
+# 8 merge master back to develop
+echo "8.3 merge master back to develop"
+# git merge <branch> <master>
+
+# 1 Create new release for tag
+echo "Step 1: Create new release for tag"
```

Comment:
```suggestion
echo "Would you like to go to the Github releases page to create a release? Y/N"
```

If Y, then open a link to the releases page. Alternatively, we might be able to use `gh` to do this? https://cli.github.com/ Maybe add a TODO for that?